File: app/src/main/java/com/boardgamegeek/auth/AccountUtils.java
package com.boardgamegeek.auth;

import android.content.Context;
import android.content.SharedPreferences;
import android.text.TextUtils;

import com.google.firebase.crashlytics.FirebaseCrashlytics;

import androidx.annotation.Nullable;
import androidx.preference.PreferenceManager;

public class AccountUtils {
	private static final String KEY_PREFIX = "account_";
	private static final String KEY_USERNAME = KEY_PREFIX + "username";
	private static final String KEY_FULL_NAME = KEY_PREFIX + "full_name";
	private static final String KEY_AVATAR_URL = KEY_PREFIX + "avatar_url";

	private AccountUtils() {
	}

	public static void clearFields(final Context context) {
		setUsername(context, null);
		setFullName(context, null);
		setAvatarUrl(context, null);
	}

	public static void setUsername(final Context context, final String username) {
		setString(context, username, KEY_USERNAME);
		if (!TextUtils.isEmpty(username))
			FirebaseCrashlytics.getInstance().setUserId(String.valueOf(username.hashCode()));
	}

	public static void setFullName(final Context context, String fullName) {
		setString(context, fullName, KEY_FULL_NAME);
	}

	public static void setAvatarUrl(final Context context, String avatarUrl) {
		setString(context, avatarUrl, KEY_AVATAR_URL);
	}

	@Nullable
	public static String getUsername(final Context context) {
		return getString(context, KEY_USERNAME);
	}

	@Nullable
	public static String getFullName(final Context context) {
		return getString(context, KEY_FULL_NAME);
	}

	@Nullable
	public static String getAvatarUrl(final Context context) {
		return getString(context, KEY_AVATAR_URL);
	}

	private static void setString(Context context, String value, String key) {
		SharedPreferences sp = getSharedPreferences(context);
		sp.edit().putString(key, value).apply();
	}

	@Nullable
	private static String getString(Context context, String key) {
		SharedPreferences sp = getSharedPreferences(context);
		return sp.getString(key, null);
	}

	private static SharedPreferences getSharedPreferences(final Context context) {
		return PreferenceManager.getDefaultSharedPreferences(context);
	}
}

File: app/src/main/java/com/boardgamegeek/auth/AccountUtils.kt
package com.boardgamegeek.auth

import android.content.Context
import com.google.firebase.crashlytics.FirebaseCrashlytics
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import com.boardgamegeek.extensions.get
import com.boardgamegeek.extensions.preferences
import com.boardgamegeek.extensions.set

object AccountUtils {
    private const val KEY_PREFIX = "account_"
    private const val KEY_USERNAME = KEY_PREFIX + "username"
    private const val KEY_FULL_NAME = KEY_PREFIX + "full_name"
    private const val KEY_AVATAR_URL = KEY_PREFIX + "avatar_url"

    fun clearFields(context: Context?) {
        setUsername(context, null)
        setFullName(context, null)
        setAvatarUrl(context, null)
    }

    fun setUsername(context: Context?, username: String?) {
        setString(context, username, KEY_USERNAME)
        if (!username.isNullOrEmpty()) FirebaseCrashlytics.getInstance().setUserId(username.hashCode().toString())
    }

    fun setFullName(context: Context?, fullName: String?) {
        setString(context, fullName, KEY_FULL_NAME)
    }

    fun setAvatarUrl(context: Context?, avatarUrl: String?) {
        setString(context, avatarUrl, KEY_AVATAR_URL)
    }

    @JvmStatic
    fun getUsername(context: Context?): String? {
        return getString(context, KEY_USERNAME)
    }

    fun getFullName(context: Context): String? {
        return getString(context, KEY_FULL_NAME)
    }

    fun getAvatarUrl(context: Context): String? {
        return getString(context, KEY_AVATAR_URL)
    }

    private fun setString(context: Context, value: String?, key: String) {
        context.preferences()[key] = value
    }

    private fun getString(context: Context, key: String): String? {
        return context.preferences()[key, ""]
    }
}
